cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -139,6 +140,53 @@ public void connectControlChannel(FullHttpRequest request, HttpResponder respond
                                    TetheringControlResponse[].class));
   }
 
+  /**
+   * Sends control commands to the client and receives program status updates from the client.
+   */
+  @POST
+  @Path("/tethering/poll/{peer}")
+  public void pollControlChannel(FullHttpRequest request, HttpResponder responder, @PathParam("peer") String peer)
+    throws IOException, NotImplementedException, BadRequestException {
+    checkTetheringServerEnabled();
+    PeerInfo peerInfo;
+    try {
+      peerInfo = store.getPeer(peer);
+    } catch (PeerNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, GSON.toJson(new TetheringErrorResponse(e.getMessage())));
+      return;
+    }
+    TetheringStatus tetheringStatus = peerInfo.getTetheringStatus();
+    store.updatePeerTimestamp(peer);
+    String messageId = processRequestContent(request);
+    List<TetheringControlMessage> controlMessages = new ArrayList<>();
+    MessageFetcher fetcher = messagingContext.getMessageFetcher();
+    TopicId topic = new TopicId(NamespaceId.SYSTEM.getNamespace(), topicPrefix + peer);
+    createTopicIfNeeded(topic);
```

Comment:
this will almost never need to be created, but is adding extra I/O to every call. It would be better to handle this in the catch TopicNotFound exception.

Reply:
fixed